### DISCUSSION TITLE
Curl fails to resolve hostname

### DISCUSSION THREAD

**Discussion by notEvil** [nextcloud/all-in-one#5557] [Questions] [answered]
Hi,

I'm facing a DNS issue which is probably not related to Nextcloud AIO but I don't know where else to put it. Since October 16th my logs are full of

**ConnectException** cURL error 28: Resolving timed out after 5000 milliseconds (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://nextcloud.my.domain/hosting/capabilities
Failed to fetch capabilities: cURL error 28: Resolving timed out after 5000 milliseconds (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://nextcloud.my.domain/hosting/capabilities 

from richdocuments. `curl https://nextcloud.my.domain/hosting/capabilities` on the host works just fine and I have no other issue with Nextcloud, including Nextcloud Office (maybe relevant: IPv6 only, no A record). My investigation so far revealed something odd:

The first execution of `curl -I -6 'https://nextcloud.my.domain/'` within `sudo docker exec -it nextcloud-aio-nextcloud bash` responds with http 302 as expected. All subsequent executions within the domain TTL window return `curl: (6) Could not resolve host: nextcloud.my.domain`. Monitoring the network with `sudo tcpdump -i br-121afda26548 'port 53'` on the host shows dns requests and responses, a single request response pair for the succeeding curl and 3 pairs for the failing ones, all with the correct response.

I'd highly appreciate any hint towards a solution, but don't expect any :)

**Comment by szaimen** [comment, marked as answer]
Hi, I fear this is most likely caused by an upstream change in the alpine image that we use internally. You might want to add a comment here: https://gitlab.alpinelinux.org/alpine/aports/-/issues/15690 or create a new issue here: https://gitlab.alpinelinux.org/alpine/aports/-/issues

**Reply by Rob-Geebivi** [reply]
I'm likely having the same issue as @notEvil...
From that same issues-15690 link, it appears a patch was merged into the 3.20 branch of Alpine, based on comments on https://gitlab.alpinelinux.org/alpine/aports/-/issues/15690 by DerLinkman on October 15th. Can this update to Alpine be included with the next AIO update to hopefully fix the issue?

**Reply by Rob-Geebivi** [reply]
To clarify my comment: I was referencing the update of the 3.20-stable git branch of Alpine that included the update to c-ares to 1.33.1 (which should supposedly fix the issue), which was done on Sep 27th. (https://gitlab.alpinelinux.org/alpine/aports/-/commit/e0e2025ea57e73d16c78170938d21c92f509a504)

I thought there was a compiled patch available right now, but it appears that the last minor patch release of Alpine, 3.20.3, was done on Sep 6th, and therefore does not include the desired fix.
Given that, it looks like the most realistic solution requires patience until Alpine 3.20.4 and/or 3.21 are released and can be incorporated into AIO.

According to Alpine's website (https://alpinelinux.org/releases/), they aim to release a major version each November, so hopefully it's not too long before it would be available to start working with. I know integration with AIO has the potential to generate its own complications, but for what it's worth, I guess you can consider this a preemptive request to update the base Alpine package used with AIO to (the not yet released) v3.20.4 or above at your earliest convenience. ;)

**Reply by szaimen** [reply]
>  I guess you can consider this a preemptive request to update the base Alpine package used with AIO to (the not yet released) v3.20.4 or above at your earliest convenience

Sounds good!

**Reply by szaimen** [reply]
It will automatically be includes as soon as the base image we are using is updated

**Reply by notEvil** [reply]
Good news, I was able to solve the issue: https://gitlab.alpinelinux.org/alpine/aports/-/issues/15690#note_472982

In short, FRITZ!Box doesn't care about Dns cookies and c-ares rejects the bad responses (`dig {domain}` shows `; COOKIE: ... (bad)`). After reconfiguration of dns to my provider servers all the issues are gone (nextcloud self check succeeds, apps "Discover" and updates work, no more errors in the logs, ...)

**Reply by Rob-Geebivi** [reply]
Interesting. Yah, after updating to NC AIO v10.3.0 (which includes the alpine updates originally thought to be the fix to the problem), I'm still having the same issue. It must be an individual DNS setup issue for me then. I'm not using FRITZ!Box or anything similar, but to keep a longer story short... I wouldn't be surprised if my internal DNS resolution is getting screwed up somewhere along the line, and that's what's causing the problem.

If this is an indicator of anything, in the web browser (Firefox) on the same machine that I have nextcloud running, I have to go to my about:config and list [mynextcloud.domain.tld] in network.dns.localDomains in order for the site to come up properly.

@notEvil - When you mention "after reconfiguration of dns to my provider servers," where and how did you do that?
Is there a way to manually adjust the DNS list in the containers?

Thanks for the help!

**Reply by notEvil** [reply]
@Rob-Geebivi I looked up the IP addresses my router is using for DNS in its web interface and used them as DNS servers for the network interface. Not sure if thats a good idea in general though. It is certainly possible to force Docker to use specific DNS settings, for instance adding `"dns": "..."` to `/etc/docker/daemon.json`. And regarding `network.dns.localDomains` DNS rebind protection is maybe a reason why it doesn't work without.

**Reply by marcolaux** [reply]
> Good news, I was able to solve the issue: https://gitlab.alpinelinux.org/alpine/aports/-/issues/15690#note_472982
> 
> In short, FRITZ!Box doesn't care about Dns cookies and c-ares rejects the bad responses (`dig {domain}` shows `; COOKIE: ... (bad)`). After reconfiguration of dns to my provider servers all the issues are gone (nextcloud self check succeeds, apps "Discover" and updates work, no more errors in the logs, ...)

Thank you very much for this hint!

I disabled DNS over TLS (DoT) in my box and it also works now.
